### PR TITLE
Implementation of the intermediate certificate pinning feature. Added…

### PR DESCRIPTION
**Pull request by mahithegeek**
… a new enum to take the mode of SSL pinning.

Modified the logic of evaluating security configuration to include the SSL pinning mode.

### CODE REVIEW

File: MASFoundation/Classes/models/MASSecurityConfiguration.h [hunk 1]
```diff
@@ -21,6 +21,27 @@
 @interface MASSecurityConfiguration : MASObject
 
 
+/**
```
Comment (ysd24) [inline MASFoundation/Classes/models/MASSecurityConfiguration.h:24]: What if the Customer wants to enable both intermediate and leaf cert pinning.
I think we need to ponder about clubbing them together with an 'OR' clause.

**Review by ysd24 — COMMENTED** [review]
What if the Customer wants to enable both intermediate and leaf cert pinning.
I think we need to ponder about clubbing them together with an 'OR' clause.

Also we need to ponder about whether this decision should be coming from server in msso_config.json ?

But otherwise looks good to me.

File: MASFoundation/Classes/models/MASSecurityConfiguration.h (hunk 1 above)
Comment (mahithegeek) [reply MASFoundation/Classes/models/MASSecurityConfiguration.h:24]: then thats a subset if you want leaf you any way get the intermediate also today....so then only pinning on intermediate has no value

**Comment by mahithegeek** [comment]
> What if the Customer wants to enable both intermediate and leaf cert pinning.
> I think we need to ponder about clubbing them together with an 'OR' clause.
> 
> Also we need to ponder about whether this decision should be coming from server in msso_config.json ?
> 
> But otherwise looks good to me.

I learned from James that msso should not be modified for this feature as pinning is not only related to MAG/OTK but also any other public server. I will forward his email on this.

File: MASFoundation/Classes/models/MASSecurityConfiguration.h (hunk 1 above)
Comment (ysd24) [reply MASFoundation/Classes/models/MASSecurityConfiguration.h:24]: What I meant it should be product management decision ?

**Review by ysd24 — APPROVED** [review]
Looks good to me.

**Pull request merged**